### PR TITLE
Parse command line double values using `.` instead of locale setting

### PR DESCRIPTION
**Pull request by pshipton**
Specifying double values on the command line should not be locale
dependent but always use '.'. This affects -XX:MaxRAMPercentage and
-XX:InitialRAMPercentage.

Fixes https://github.com/eclipse-openj9/openj9/issues/13057

Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins compile zlinux,win,aix,osx jdk8

**Comment by pshipton** [comment]
jenkins compile zlinux,win,aix,osx jdk8

**Comment by pshipton** [comment]
jenkins compile zlinux,win,aix,osx jdk8

**Comment by pshipton** [comment]
jenkins compile zlinux,win,aix,osx jdk8

**Comment by pshipton** [comment]
jenkins compile zlinux,win,aix,osx jdk8

**Comment by pshipton** [comment]
jenkins compile zlinux,win,aix,osx jdk8

**Comment by pshipton** [comment]
jenkins compile win,osx jdk8

**Comment by pshipton** [comment]
jenkins compile osx jdk8

File: runtime/util_core/j9argscan.c [hunk 1]
```diff
@@ -111,8 +115,25 @@ void scan_failed(J9PortLibrary * portLibrary, const char* module, const char *sc
...
+	/* AIX 7.1+ has support for newlocale() */
+	locale_t clocale = newlocale(LC_ALL_MASK, "C", NULL);
+	locale_t orglocale = uselocale(clocale);
+	*result = strtod(*scan_start, &endPtr);
+	uselocale(orglocale);
+	freelocale(clocale);
+#else  /* defined(WINDOWS) */
+	/* No support on AIX 6.1 or z/OS so far, but testing shows strtod() isn't locale dependent */
 	*result = strtod(*scan_start, &endPtr);
+#endif /* defined(WINDOWS) */
```
Comment (keithc-ca) [inline runtime/util_core/j9argscan.c:136]: Each of `_create_locale()` and `newlocale()` can fail, returning `NULL`. That should be checked, in which case, we could revert to using `strtod()`.

nit: extra space after `#else`

File: runtime/util_core/j9argscan.c [hunk 2]
```diff
@@ -111,8 +115,25 @@ void scan_failed(J9PortLibrary * portLibrary, const char* module, const char *sc
...
 {
 	char *endPtr = NULL;
-
+#if defined(WINDOWS)
+	_locale_t clocale = _create_locale(LC_ALL, "C");
+	*result = _strtod_l(*scan_start, &endPtr, clocale);
+	_free_locale(clocale);
+#elif defined(OSX) /* defined(WINDOWS) */
+	locale_t clocale = newlocale(LC_ALL_MASK, "C", NULL);
+	*result = strtod_l(*scan_start, &endPtr, clocale);
```
Comment (keithc-ca) [inline runtime/util_core/j9argscan.c:124]: `strtod_l()` appears to be a BSD extension on OSX, why not just use the Linux pattern on OSX if we're going to depend on BSD-isms?

File: runtime/util_core/j9argscan.c (hunk 1 above)
Comment (pshipton) [reply runtime/util_core/j9argscan.c]: Updated.

File: runtime/util_core/j9argscan.c (hunk 2 above)
Comment (pshipton) [reply runtime/util_core/j9argscan.c]: Sure, updated.

File: runtime/util_core/j9argscan.c [hunk 3]
```diff
@@ -111,8 +115,29 @@ void scan_failed(J9PortLibrary * portLibrary, const char* module, const char *sc
...
+	if (NULL != clocale) {
+		locale_t orglocale = uselocale(clocale);
+		*result = strtod(*scan_start, &endPtr);
+		uselocale(orglocale);
+		freelocale(clocale);
+	} else {
+		*result = strtod(*scan_start, &endPtr);
+	}
+#else /* defined(WINDOWS) */
+	/* No support on AIX 6.1 or z/OS so far, but testing shows strtod() isn't locale dependent */
```
Comment (pshipton) [inline runtime/util_core/j9argscan.c:138]: Realized I tested it incorrectly. z/OS at least is locale specific so I need another solution.

**Comment by pshipton** [comment]
jenkins compile zlinux,win,aix,osx jdk8

**Comment by pshipton** [comment]
jenkins compile zlinux,win,aix,osx jdk8

**Comment by pshipton** [comment]
jenkins compile win jdk8

File: runtime/util_core/j9argscan.c [hunk 4]
```diff
@@ -111,8 +118,57 @@ void scan_failed(J9PortLibrary * portLibrary, const char* module, const char *sc
...
+		locale_t orglocale = uselocale(clocale);
+		*result = strtod(*scan_start, &endPtr);
+		uselocale(orglocale);
+		freelocale(clocale);
+		bruteParse = 0;
+	}
+#endif /* defined(OMR_OS_WINDOWS) */
+
+	/* There is no per thread locale support on AIX 6.1 or z/OS so far. */
+	if (1 == bruteParse) {
```
Comment (keithc-ca) [inline runtime/util_core/j9argscan.c:142]: Suggest `0 != bruteParse`.

File: runtime/util_core/j9argscan.c [hunk 5]
```diff
@@ -111,8 +118,57 @@ void scan_failed(J9PortLibrary * portLibrary, const char* module, const char *sc
...
+#if defined(OMR_OS_WINDOWS)
+		struct lconv *linfo = localeconv();
+		char *localeRadix = linfo->decimal_point;
+#else /* defined(OMR_OS_WINDOWS) */
+		char *localeRadix = nl_langinfo(RADIXCHAR);
+#endif /* defined(OMR_OS_WINDOWS) */
+		if ((NULL != localeRadix) && ('.' != *localeRadix)) {
+			char *radixPos = NULL;
+			size_t len = strlen(*scan_start);
+			if ((len + 1) > sizeof(buffer)) {
```
Comment (keithc-ca) [inline runtime/util_core/j9argscan.c:156]: Suggest `len >= sizeof(buffer)`.

File: runtime/util_core/j9argscan.c (hunk 4 above)
Comment (pshipton) [reply runtime/util_core/j9argscan.c]: Updated

File: runtime/util_core/j9argscan.c (hunk 5 above)
Comment (pshipton) [reply runtime/util_core/j9argscan.c]: Updated

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**